blobstore/README.md

```diff
@@ -7,12 +7,51 @@ files. It is like a simpler S3. All operations fall under these three HTTP
 verbs:
 
  * PUT: write a file: `curl -X PUT -T /path/to/local/file
-   http://blobstorehost/path/to/remote/file`
- * GET: read a file: `curl http://blobstorehost/path/to/remote/file`
+   http://blobstore.discoverd/path/to/remote/file`
+ * GET: read a file: `curl http://blobstore.discoverd/path/to/remote/file`
  * DELETE: delete a file: `curl -X DELETE
-   http://blobstorehost/path/to/remote/file`
+   http://blobstore.discoverd/path/to/remote/file`
+
+Files can be copied by setting the `X-Blobstore-Copy-From` header in a PUT
+request:
+
+```shell
+# create /file1.txt
+echo data | curl -X PUT --data-binary @- http://blobstore.discoverd/file1.txt
+
+# copy it to /file2.txt
+curl -X PUT --header "X-Blobstore-Copy-From: /file1.txt" http://blobstore.discoverd/file2.txt
+
+# read /file2.txt
+curl http://blobstore.discoverd/file2.txt
+data
+
+```
+
+
+Parent directories are automatically created, and directory indexes can be
+accessed with a GET request to the root path with the `dir` query param:
+
+```shell
+# create 4 files in different directories
+curl -X PUT --data-binary "data" http://blobstore.discoverd/foo.txt
+curl -X PUT --data-binary "data" http://blobstore.discoverd/dir1/foo.txt
+curl -X PUT --data-binary "data" http://blobstore.discoverd/dir2/foo.txt
+curl -X PUT --data-binary "data" http://blobstore.discoverd/dir3/foo.txt
+
+# list all top-level files and directories
+curl http://blobstore.discoverd/
+["/dir1/","/dir2/","/dir3/","/foo.txt"]
+
+# list files in /dir1
+curl http://blobstore.discoverd/?dir=/dir1
+["/dir1/foo.txt"]
+
+# list files in /dir2
+curl http://blobstore.discoverd/?dir=/dir2
```

Comment:
So my first pass at this did have `GET /dir` return a directory listing but this lead to the need to handle the following:

* don't allow both `/dir` and `/dir/foo.txt` to be regular files (since then `GET /dir` would be ambiguous)
* update postgres to know whether `GET /dir` should lookup a file or do a prefix search

The solution I came up with was to model directories explicitly in postgres by adding a `dir` column and adding a `MkdirAll` method (see [this commit](https://github.com/flynn/flynn/commit/d7a76e331aff67b7e2446bcc6d02c996b57f67b8)), but then I realised that S3 doesn't model directories like this, so if we want the blobstore to become a gateway to S3 like services, we should probably stick closer to their API, hence why I went with `GET /` for directory listings and `GET /path` to get files.

It is also much simpler (although if you think there is a simpler way to accomplish your approach without the complexity in my commit, let me know).